CRM/Twingle/Form/Profile.php

```diff
@@ -139,61 +145,62 @@ public function buildQuickForm() {
       $this->_op = 'create';
     }
 
-    // Verify that a profile with the given name exists.
-    $profile_name = CRM_Utils_Request::retrieve('name', 'String', $this);
-    if (!$this->profile = CRM_Twingle_Profile::getProfile($profile_name)) {
-      $profile_name = NULL;
+    // Verify that a profile with the given id exists.
+
+    if ($this->_op != 'copy') {
+      $this->profile_id = CRM_Utils_Request::retrieve('id', 'Int', $this);
+      $this->profile = CRM_Twingle_Profile::getProfile($this->profile_id);
     }
 
     // Set redirect destination.
     $this->controller->_destination = CRM_Utils_System::url('civicrm/admin/settings/twingle/profiles', 'reset=1');
 
     switch ($this->_op) {
       case 'delete':
-        if ($profile_name) {
-          CRM_Utils_System::setTitle(E::ts('Delete Twingle API profile <em>%1</em>', array(1 => $profile_name)));
-          $this->addButtons(array(
-            array(
+        if ($this->profile_id) {
+          CRM_Utils_System::setTitle(E::ts('Delete Twingle API profile <em>%1</em>', [1 => $this->profile->getName()]));
+          $this->addButtons([
+            [
               'type' => 'submit',
-              'name' => ($profile_name == 'default' ? E::ts('Reset') : E::ts('Delete')),
+              'name' => ($this->profile->getName() == 'default' ? E::ts('Reset') : E::ts('Delete')),
               'isDefault' => TRUE,
-            ),
-          ));
+            ],
+          ]);
         }
         parent::buildQuickForm();
         return;
-      case 'edit':
-        // When editing without a valid profile name, edit the default profile.
-        if (!$profile_name) {
-          $profile_name = 'default';
-          $this->profile = CRM_Twingle_Profile::getProfile($profile_name);
-        }
-        CRM_Utils_System::setTitle(E::ts('Edit Twingle API profile <em>%1</em>', array(1 => $this->profile->getName())));
-        break;
+
       case 'copy':
         // Retrieve the source profile name.
-        $profile_name = CRM_Utils_Request::retrieve('source_name', 'String', $this);
-        // When copying without a valid profile name, copy the default profile.
-        if (!$profile_name) {
-          $profile_name = 'default';
+        $source_id = CRM_Utils_Request::retrieve('source_id', 'Int', $this);
+        // When copying without a valid profile id, copy the default profile.
+        if (!$source_id) {
+          $this->profile = CRM_Twingle_Profile::createDefaultProfile();
+        } else {
+          $source_profile = CRM_Twingle_Profile::getProfile($source_id);
```

Comment:
This should be validated, as we can't be sure the passed `source_profile` value ist actually a valid profile.